server/channels/store/sqlstore/property_field_store.go

```diff
@@ -0,0 +1,339 @@
+			&field.TargetType,
+			&field.CreateAt,
+			&field.UpdateAt,
+			&field.DeleteAt,
+		)
+		if err != nil {
+			return nil, err
+		}
+
+		if err := json.Unmarshal([]byte(attrsJSON), &field.Attrs); err != nil {
+			return nil, errors.Wrap(err, "property_fields_from_rows_unmarshal_attrs")
+		}
+
+		results = append(results, &field)
+	}
+
+	return results, nil
+}
+
+func propertyFieldFromRows(rows *sql.Rows) (*model.PropertyField, error) {
+	fields, err := propertyFieldsFromRows(rows)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(fields) > 0 {
+		return fields[0], nil
+	}
+
+	return nil, sql.ErrNoRows
+}
+
+type SqlPropertyFieldStore struct {
+	*SqlStore
+}
+
+func newPropertyFieldStore(sqlStore *SqlStore) store.PropertyFieldStore {
+	return &SqlPropertyFieldStore{sqlStore}
+}
+
+func (s *SqlPropertyFieldStore) Create(field *model.PropertyField) (*model.PropertyField, error) {
+	if field.ID != "" {
+		return nil, store.NewErrInvalidInput("PropertyField", "id", field.ID)
+	}
+
+	field.PreSave()
+
+	if err := field.IsValid(); err != nil {
+		return nil, errors.Wrap(err, "property_field_create_isvalid")
+	}
+
+	insertMap, err := s.propertyFieldToInsertMap(field)
+	if err != nil {
+		return nil, err
+	}
+
+	queryString, args, err := s.getQueryBuilder().
+		Insert("PropertyFields").
+		SetMap(insertMap).
+		ToSql()
+	if err != nil {
+		return nil, errors.Wrap(err, "property_field_create_tosql")
+	}
+
+	if _, err := s.GetMaster().Exec(queryString, args...); err != nil {
+		return nil, errors.Wrap(err, "property_field_create_insert")
+	}
+
+	return field, nil
+}
+
+func (s *SqlPropertyFieldStore) Get(id string) (*model.PropertyField, error) {
+	queryString, args, err := s.getQueryBuilder().
+		Select(propertyFieldColumns...).
+		From("PropertyFields").
+		Where(sq.Eq{"id": id}).
+		ToSql()
+	if err != nil {
+		return nil, errors.Wrap(err, "property_field_get_tosql")
+	}
+
+	rows, err := s.GetReplica().Query(queryString, args...)
+	if err != nil {
+		return nil, errors.Wrap(err, "property_field_get_select")
+	}
+	defer rows.Close()
+
+	field, err := propertyFieldFromRows(rows)
+	if err != nil {
+		return nil, errors.Wrap(err, "property_field_get_propertyfieldfromrows")
+	}
+
+	return field, nil
+}
+
+func (s *SqlPropertyFieldStore) GetMany(ids []string) ([]*model.PropertyField, error) {
+	queryString, args, err := s.getQueryBuilder().
+		Select(propertyFieldColumns...).
+		From("PropertyFields").
+		Where(sq.Eq{"id": ids}).
+		ToSql()
+	if err != nil {
+		return nil, errors.Wrap(err, "property_field_get_many_tosql")
+	}
+
+	rows, err := s.GetReplica().Query(queryString, args...)
+	if err != nil {
+		return nil, errors.Wrap(err, "property_field_get_many_query")
+	}
+	defer rows.Close()
+
+	fields, err := propertyFieldsFromRows(rows)
+	if err != nil {
+		return nil, errors.Wrap(err, "property_field_get_many_propertyfieldfromrows")
+	}
+
+	if len(fields) < len(ids) {
+		return nil, errors.New("property_field_get_many_missmatch_results")
```

Comment:
`property_field_get_many_missmatch_results` barely has any context. Let's try to have a more user friendly error: `Number of ids got: %d is fewer than number of ids passed: %d`.

Reply:
Makes sense, fixed in 75c75982d1